File: templates/python/fr/text/code/stepUp2.py
# Fichier: Jeton PickingKarel.py
# -----------------------------
# Le programme Jeton Jeton définit un " main "
# fonction avec trois commandes. Ces commandes provoquent
# Karel à avancer avancer d'un bloc, prendre un
# jeton puis avancer devant le coin suivant.
from karel.stanfordkarel import *
def main():
   avancer()
   ramasser_jeton()
   avancer()
   tourner_gauche()
   avancer()
   tourner_droite()
   avancer()
   avancer()
   laisser_jeton()
   avancer()

def tourner_droite():
   tourner_gauche()
   tourner_gauche()
   tourner_gauche()

File: templates/java/fr/text/code/stepUp2.java
/*
 * Fichier: JetonRamasserKarel.java
 * --------------------------------
 * Le programme JetonRamasserKarel définit une méthod
 * e " run " à trois commandes. Ces commandes entraîn
 * ent Karel avancer avant d'un bloc, jeton un jeton
 * , puis avancer avant jusqu'au prochain virage.
 */
import stanford.karel.*;
public class JetonRamasserKarel extends Karel {
   public void run() {
      avancer();
      ramasserJeton();
      avancer();
      tournerGauche();
      avancer();
      tournerDroite();
      avancer();
      avancer();
      laisserJeton();
      avancer();
   }

   /**
    * Tourne Karel à 90 degrés vers la droite.
    */
   private void tournerDroite() {
      tournerGauche();
      tournerGauche();
      tournerGauche();
   }
}